src/app/view/components/dot-workflow-task-detail/services/dot-workflow-task-detail.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Observable';
+import { mergeMap, map, filter } from 'rxjs/operators';
+import { DotMenuService } from '../../../../api/services/dot-menu.service';
+import { of } from 'rxjs/observable/of';
+
+export interface DotEditorAction {
```

Comment:
You are duplicating this interface, you need to create one for the workflows `DotTaskAction` maybe?, also it could be simpler header and id.